update_twiki.py

```diff
@@ -0,0 +1,109 @@
+"""
+import sys
+import sqlite3
+import logging
+import numpy as np
+sys.path.append('/afs/cern.ch/cms/PPD/PdmV/tools/McM/')
+from rest import McM
+#XSDB wrapper
+from request_wrapper import RequestWrapper
+#XSDB pycurl requester instance
+xsdb_request = RequestWrapper()
+# McM instance
+mcm = McM(dev=False, debug=False, cookie='cookie.txt')
+# Logger
+logging.basicConfig(format='[%(asctime)s][%(levelname)s] %(message)s', level=logging.INFO)
+logger = logging.getLogger()
+
+def create_table():
+    conn = sqlite3.connect('twiki.db')
+    c = conn.cursor()
+    # Create table if it does not exist
+    c.execute('''CREATE TABLE IF NOT EXISTS twiki_samples
+                 (prepid text NOT NULL,
+                  dataset text NOT NULL,
+                  extension text,
+                  total_events integer NOT NULL,
+                  campaign text NOT NULL,
+                  resp_group text NOT NULL,
+                  cross_section float NOT NULL,
+                  fraction_negative_weight float NOT NULL,
+                  target_num_events real NOT NULL   )''')
+
+    # Clear the table
+    c.execute('DELETE FROM `twiki_samples`')
+    conn.commit()
+    conn.close()
+def insert_update(twiki_request, campaign, cross_section, frac_neg_wgts, target_num_events):
+    conn = sqlite3.connect('twiki.db')
+    c = conn.cursor()
+    logger.info('Inserting %s (%s) %s %s %s', twiki_request['dataset_name'], twiki_request['prepid'], cross_section, frac_neg_wgts, target_num_events)
+    c.execute('INSERT INTO twiki_samples VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?)',
+              [twiki_request['prepid'],
+               twiki_request['dataset_name'],
+               twiki_request['extension'],
+               twiki_request['total_events'],
+               campaign,
+               twiki_request['prepid'].split('-')[0],
+               cross_section,
+               frac_neg_wgts,
+               target_num_events])
+
+    conn.commit()
+    conn.close()
+def operations():
+    query_ul18 = 'member_of_campaign=RunIISummer19UL18MiniAOD'
+    query_ul16 = 'member_of_campaign=RunIISummer19UL16MiniAOD'
+    query_ul17 = 'member_of_campaign=RunIISummer19UL17MiniAOD'
+    requests_ul17 = mcm.get('requests', query=query_ul17)
+    requests_ul18 = mcm.get('requests', query=query_ul18)
+    requests_ul16 = mcm.get('requests', query=query_ul16)
+    missing_ul18 = []
+    missing_ul16 = []
+    ul17_dataset_names = [x['dataset_name'] for x in requests_ul17]
+    ul16_dataset_names = [x['dataset_name'] for x in requests_ul16]
+    ul18_dataset_names = [x['dataset_name'] for x in requests_ul18]
+    missing_ul18 = np.setdiff1d(ul17_dataset_names, ul18_dataset_names)
+    missing_ul16 = np.setdiff1d(ul17_dataset_names, ul16_dataset_names)
+
+    # Get all needed requests
+    total_events_threshold = 0#20000000
+    for twiki_request in requests_ul17:
+        if twiki_request['total_events'] > total_events_threshold:
+            campaign = 'RunIISummer19UL17'
```

Comment:
`campaign` set here and in two `if` statements below seems to be useless as it is overwritten in lines 100 and 104.

Reply:
Here I set it so I can get out of the loop earlier for efficiency, but then I noticed that for the samples with the same name I will overwrite the campaign name if I don't put the if statements on lines 100 and 104 :)

Reply:
Aha, I see, I totally misread these `ifs` during last preview. I thought they were `if-elif-elif` and not three separate `if`.

However, I still think that a separate `campaign` variable here is not needed, except to store `if` result which then leads to string comparison. This can be replaced with this to get rid of `campaign` all together:
```
if twiki_request['dataset_name'] not in (missing_ul16 + missing_ul18):
    continue
```
This joins two sets together during every iteration which is sub-optimal too, so that can be moved outside the loop to something like:
```
missing_ul16_and_ul18 = missing_ul16 + missing_ul18
```
And then would look like:
```
if twiki_request['dataset_name'] not in missing_ul16_and_ul18:
    continue
```
However, if you think your if is good enough, fine, you can leave it as it is :+1: